### PR TITLE
dvclive callback: warn instead of fail when logging non-scalars

### PR DESCRIPTION
**Pull request by dberenbaum**
# What does this PR do?

Fixes https://github.com/huggingface/transformers/pull/27352#issuecomment-1819131456. This will warn instead of fail when trying to log non-scalars as metrics.

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [x] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

@muellerz Could you please take a look?

### CODE REVIEW

**Comment by dberenbaum** [comment]
@muellerz This makes the tests pass, but I'm not sure if it's intended that the test here logs the learning rate as a list rather than as a scalar (which will fail under several of the existing loggers, but only with a warning like in this PR). 

https://github.com/huggingface/transformers/blob/e4280d650c579a87f645d1f4a4535feb27c49804/tests/trainer/test_trainer.py#L675

`self.lr_scheduler._last_lr` is a [list](https://github.com/pytorch/pytorch/blob/140c54e6ccc5e97f1b7f1e0fcd3d8c6af7dd2ab2/torch/optim/lr_scheduler.py#L167). Should a scalar value be extracted like `self.lr_scheduler._last_lr[0]`? That's the value being tested later as `["learning_rate"][0]`:

https://github.com/huggingface/transformers/blob/f31af3927f4091f5fb8126c77a0addebd4c1fe94/tests/trainer/test_trainer.py#L699-L712

Everywhere else in the codebase, it looks like a scalar is extracted:

https://github.com/huggingface/transformers/blob/f31af3927f4091f5fb8126c77a0addebd4c1fe94/src/transformers/trainer_pt_utils.py#L847-L867

https://github.com/huggingface/transformers/blob/f31af3927f4091f5fb8126c77a0addebd4c1fe94/examples/legacy/pytorch-lightning/run_glue.py#L46

**Comment by muellerzr** [comment]
In the future it's @muellerzr @dberenbaum, don't want to be pinging random people :) 

Yes, let's go with `[0]` as the one being extracted/the scalar.

**Review by muellerzr — APPROVED** [review]
Thanks for the fix! If we can log just the scalar too as part of this that would be great too. Otherwise this PR LG2M. Appreciate the quickfix :)

**Comment by dberenbaum** [comment]
@muellerzr Apologies to you and the other person who was pinged here Zach! Added the change to the test in the last commit. The current test failures look unrelated.

**Review by ArthurZucker — APPROVED** [review]
Thanks 😉
